dali/python/nvidia/dali/data_node.py

```diff
@@ -122,6 +139,63 @@ def __bool__(self):
                 " represent computations in DALI Pipeline see the \"Mathematical Expressions\""
                 " section of DALI documentation."))
 
+    def __getitem__(self, val):
+        idxs = []
+        new_axes = []
+        new_axis_names = []
+
+        # returns True if this index adds a new output dimension
+        def process_index(idx, dim):
+            if idx is None:
+                idxs.append((None, None, None, None))
+                return True
+            elif isinstance(idx, slice):
+                if idx.step is not None and idx.step != 1:
+                    raise NotImplementedError("Slicing with non-unit step is not implemented.")
```

Comment:
Do we need it here too or the native side is sufficient?

Reply:
And how about negative step (`-1`)?

Reply:
Not supported yet.